tests/integration/commands/test_init.py

```diff
@@ -156,3 +165,17 @@ def test_init_nonoverlapping_file(app, capsys, monkeypatch):
     assert return_code == os.EX_OK
     assert expected_output in stdout
     assert pathlib.Path("simple-file").is_file()
+
+
+@pytest.mark.usefixtures("fake_template_dirs")
+def test_init_invalid_directory(app, monkeypatch, tmp_path):
+    """A default name is used if the project dir is not a valid project name."""
+    invalid_dir = tmp_path / "invalid--name"
+    invalid_dir.mkdir()
+    monkeypatch.chdir(invalid_dir)
+
+    monkeypatch.setattr("sys.argv", ["testcraft", "init", "--profile", "simple"])
+    app.run()
```

Comment:
As a general question. Should we check exit code in our integration tests?

Reply:
Yes! Specially in cases like this, it was definitely possible that the command did things *and* failed, which would make this a false-positive
updated